### ISSUE TITLE
Dendron views and some command doesn't work as expected with remote vault

### ISSUE THREAD

**Issue by doylejg** [dendronhq/dendron#3301]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [X] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [ ] Lookup.  Dendron's Lookup Command
- [ ] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [X] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [ ] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

When I try and create a schema from a note I get the error message: 
```log
{"level":30,"time":1658902019879,"pid":25160,"hostname":"DESKTOP-M25NKFA","name":"dendron","ctx":"refreshBacklinksChangeActiveTextEditor"}
{"level":50,"time":1658902023530,"pid":25160,"hostname":"DESKTOP-M25NKFA","name":"dendron","ctx":"dendron.createSchemaFromHierarchy:run","error":{"name":"DendronError","status":"unknown","payload":"{\"wsRoot\":\"c:\\\\Users\\\\jonat\\\\Dendron\",\"fsPath\":\"c:\\\\Users\\\\jonat\\\\Dendron\\\\dependencies\\\\github.com\\\\doylejg\\\\work\",\"vaults\":[{\"fsPath\":\"dependencies/github.com/doylejg/work\",\"remote\":{\"type\":\"git\",\"url\":\"git@github.com:doylejg/work.git\"},\"name\":\"work\"}],\"normPath\":\"dependencies\\\\github.com\\\\doylejg\\\\work\",\"msg\":\"no vault found\"}"}}
```

I cannot view graphs either: 
```log
{"level":30,"time":1658902087502,"pid":25160,"hostname":"DESKTOP-M25NKFA","name":"dendron","ctx":"WindowWatcher:onDidChangeVisibleTextEditors","editorPaths":["tasks","c:\\Users\\jonat\\Dendron\\dependencies\\github.com\\doylejg\\work\\daily.journal.2022.07.26.md"]}
{"level":30,"time":1658902087519,"pid":25160,"hostname":"DESKTOP-M25NKFA","name":"dendron","ctx":"CalendarView:openTextDocument","uri":"c:\\Users\\jonat\\Dendron\\dependencies\\github.com\\doylejg\\work\\daily.journal.2022.07.26.md","msg":"not in workspace"}
{"level":30,"time":1658902087519,"pid":25160,"hostname":"DESKTOP-M25NKFA","name":"dendron","ctx":"refreshBacklinksChangeActiveTextEditor"}
{"level":30,"time":1658902091856,"pid":25160,"hostname":"DESKTOP-M25NKFA","name":"dendron","ctx":"dendron.showNoteGraph:run","msg":"pre-execute"}
```

## To Reproduce

1. Setup workspace with only remote git vault
2. Click on graph or schema

## Expected behavior

I expected local graphs and schema

## Desktop (please complete the following information)

- OS: Windows 10
- Versions (all that apply to bug):
  - Dendron Extension: v0.105.0
  - VS Code version: 1.69.2
  - Dendron CLI: ?

## Additional context

Joshi reproduced the issue on discord

## Dendron Log file

Please attach the output of `> Dendron:Dev: Open Logs` here

```yaml
version: 5
dev:
    enablePreviewV2: true
    enableSelfContainedVaults: true
commands:
    lookup:
        note:
            selectionMode: extract
            confirmVaultOnCreate: true
            vaultSelectionModeOnCreate: smart
            leaveTrace: false
            bubbleUpCreateNew: true
            fuzzThreshold: 0.2
    randomNote: {}
    insertNoteLink:
        aliasMode: none
        enableMultiSelect: false
    insertNoteIndex:
        enableMarker: false
    copyNoteLink: {}
    templateHierarchy: template
workspace:
    vaults:
        -
            fsPath: dependencies/github.com/doylejg/work
            remote:
                type: git
                url: git@github.com:doylejg/work.git
            name: work
    journal:
        dailyDomain: daily
        name: journal
        dateFormat: y.MM.dd
        addBehavior: childOfDomain
    scratch:
        name: scratch
        dateFormat: y.MM.dd.HHmmss
        addBehavior: asOwnDomain
    task:
        name: task
        dateFormat: y.MM.dd
        addBehavior: asOwnDomain
        statusSymbols:
            '': ' '
            wip: w
            done: x
            assigned: a
            moved: m
            blocked: b
            delegated: l
            dropped: d
            pending: 'y'
        taskCompleteStatus:
            - done
            - x
        prioritySymbols:
            H: high
            M: medium
            L: low
        todoIntegration: false
        createTaskSelectionType: selection2link
    graph:
        zoomSpeed: 1
        createStub: false
    enableAutoCreateOnDefinition: false
    enableHandlebarTemplates: true
    enableXVaultWikiLink: false
    enableRemoteVaultInit: true
    enableUserTags: true
    enableHashTags: true
    workspaceVaultSyncMode: noCommit
    enableAutoFoldFrontmatter: false
    enableEditorDecorations: true
    maxPreviewsCached: 10
    maxNoteLength: 204800
    enableFullHierarchyNoteTitle: false
    enableSmartRefs: true
    seeds:
        dendron.templates: {}
preview:
    enableFMTitle: true
    enableNoteTitleForLink: true
    enableFrontmatterTags: true
    enableHashesForFMTags: false
    enableMermaid: true
    enablePrettyRefs: true
    enableKatex: true
    automaticallyShowPreview: false
publishing:
    enableFMTitle: true
    enableNoteTitleForLink: true
    enableMermaid: true
    enablePrettyRefs: true
    enableKatex: true
    copyAssets: true
    siteHierarchies:
        - root
    writeStubs: false
    siteRootDir: docs
    seo:
        title: Dendron
        description: Personal Knowledge Space
    github:
        enableEditLink: true
        editLinkText: Edit this page on GitHub
        editBranch: main
        editViewMode: tree
    enableSiteLastModified: true
    enableFrontmatterTags: true
    enableHashesForFMTags: false
    enableRandomlyColoredTags: true
    enableTaskNotes: true
    enablePrettyLinks: true
```

**Comment by Harshita-mindfire** [comment]
Thanks @doylejg for reporting the issue. It looks like the issue extends to other views aswell: `Preview`, `Backlinks` and command: `Create Schema From Hierarchy` also fails with `no vault found` error

**Comment by Harshita-mindfire** [comment]
This should be fixed with Dendron v0.106. Feel free to re-open if not the case

**Comment by Harshita-mindfire** [comment]
This should be fixed after Dendron v0.107. Feel free to ping here if that is not the case.
